### PR TITLE
[Security] Upgrade app-project to next@13.5

### PR DESCRIPTION
**Pull request by eatyourgreens**
This PR will close [a low severity security alert](https://github.com/zooniverse/front-end-monorepo/security/dependabot/81). 

- [x] upgrade `next` to v13.
- [x] convert `router-context` to `router-context.shared-runtime`.
- [x] upgrade `next/link` to the v13 Link API.
- [x] fix build warnings.
- [x] fix failing tests.
- [x] identify and fix any broken client-side `Link` components.

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
- app-project

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
- What user actions should my reviewer step through to review this PR?
- Which storybook stories should be reviewed?
- Are there plans for follow up PR’s to further fix this bug or develop this feature?

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Maintenance
- [ ] If not from dependabot, the PR creator has described the update (major, minor, or patch version, changelog)

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Something to check: will the app server start, on Kubernetes, with Next 13 (#5428.)

If not, we'll need to run the app with `next start`, which will prevent us from testing Panoptes auth from local builds.

**Comment by goplayoutside3** [comment]
Testing Panoptes auth locally is still possible with `yarn dev` in this scenario though, is that correct?

**Comment by eatyourgreens** [comment]
>Testing Panoptes auth locally is still possible with `yarn dev` in this scenario though, is that correct?

It is, but development mode disables client-side routing (among other things) so doesn't really test how the built app will behave in production.

**Comment by goplayoutside3** [comment]
Do you know where in the Next.js docs it describes features disabled in development mode? I briefly looked, but couldn't find the docs section. No worries if not, I'll continue to research.

**Comment by eatyourgreens** [comment]
I can't remember either but I do remember accidentally merging and deploying a PR once, which we'd tested with `yarn dev`, but which then broke when run with `next build && next start`.

**Comment by goplayoutside3** [comment]
Agreed it would be good to know what PR that was too, for reference on what features are not accurately testable via dev mode.

**Comment by eatyourgreens** [comment]
I think it might have been one of the PRs that originally added client-side routing to the project app.

**Comment by eatyourgreens** [comment]
Here you go. This is one of a series of PRs that worked with `yarn dev` but failed in production.
- https://github.com/zooniverse/front-end-monorepo/pull/1048

**Comment by eatyourgreens** [comment]
`yarn dev` also enables Fast Refresh, which, of course, won’t be running on the production app.

**Comment by eatyourgreens** [comment]
53 failing tests, but this should now build and deploy to kubernetes without problems.

**Comment by eatyourgreens** [comment]
Here's a couple of test projects:
- https://fe-project-branch.preview.zooniverse.org/projects/zooniverse/gravity-spy
- https://fe-project-branch.preview.zooniverse.org/projects/printmigrationnetwork/print

**Comment by eatyourgreens** [comment]
Only 11 failing tests now.

**Comment by eatyourgreens** [comment]
`WorkflowMenuModal` enzyme tests still throw errors. Other than that, this is ready.

**Comment by eatyourgreens** [comment]
I've also opened #5678, which is a duplicate of this branch but upgrades `next` to 14.0.3 across the monorepo.

**Comment by eatyourgreens** [comment]
I've skipped the `WorkflowMenuModal` tests until #5370 is fixed.

**Comment by goplayoutside3** [comment]
Did a quick glance at the deployed version of this branch `next-13`, and translations don't work as expected. Go to https://fe-project-branch.preview.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify and use the locale switcher to change to Spanish. As the classify page re-renders, the workflow selection modal appears unexpectedly and I see translation key names instead of their translation string values.

Is the latest of this branch deployed to fe-project-branch? If so, this PR need more work on translation before it can be fully reviewed.

**Comment by goplayoutside3** [comment]
I'll note too that going directly to a non-English project page url results in 404, so something in this upgrade to Next.js 13 is interfering with our established locale=subpath convention.

https://fe-project-branch.preview.zooniverse.org/projects/fr/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235

**Comment by eatyourgreens** [comment]
I rewrote the URL rewrites (although I don't know why they broke.) This link should work now.
https://fe-project-branch.preview.zooniverse.org/projects/fr/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235

File: packages/app-project/next.config.js
Comment (goplayoutside3) [inline packages/app-project/next.config.js:60]: Is `optimizePackageImports` the updated syntax for `modularizeImports`? Why not include `lodash` in the new array?

**Review by goplayoutside3 — APPROVED** [review]
This looks good! I'll do a bit more manual testing once merged, too. Please let me know if I need to hit merge with approval - I'm not sure how authorship vs recently changed Github permissions play together.

File: packages/app-project/next.config.js
Comment (eatyourgreens) [reply packages/app-project/next.config.js:60]: `lodash` is included by default. 

https://nextjs.org/docs/app/api-reference/next-config-js/optimizePackageImports

Comment (eatyourgreens) [reply packages/app-project/next.config.js:60]: This blog post explains how Next.js handles barrel files and goes into details. 

https://vercel.com/blog/how-we-optimized-package-imports-in-next-js

One takeaway I got from that is that it would probably make sense to remove barrel files from the libraries. They slow down imports and make the bundle bigger than it should be.

**Comment by eatyourgreens** [comment]
I can’t merge Zooniverse PRs, so you’ll need to merge this for me. I’ve been kicked out of the Zooniverse GitHub teams. 😐

File: packages/app-project/next.config.js
Comment (goplayoutside3) [reply packages/app-project/next.config.js:60]: I was wondering about this too! There's lots of barrel imports in lib-classifier for example.

Comment (eatyourgreens) [reply packages/app-project/next.config.js:60]: Yeah, I used them a lot, in the classifier, to make code easier to organise but they may have been a bad idea.

**Comment by eatyourgreens** [comment]
🎉

**Pull request merged**
